helpers.c

```diff
@@ -313,4 +313,166 @@ hsl_to_rgb_float (float *h_, float *s_, float *l_)
   *l_ = b;
 }
 
+// RGB to RYB
+// adapted from http://www.deathbysoftware.com/colors/index.html
+	/**********************************************************************************************
+	 * Given a RYB color, calculate the RGB color.  This code was taken from:
+	 * 
+	 * @param iRed     The current red value.
+	 * @param iYellow  The current yellow value.
+	 * @param iBlue    The current blue value.
+	 * 
+	 * http://www.insanit.net/tag/rgb-to-ryb/
+	 * 
+	 * Author: Arah J. Leonard
+	 * Copyright 01AUG09
+	 * Distributed under the LGPL - http://www.gnu.org/copyleft/lesser.html
+	 * ALSO distributed under the The MIT License from the Open Source Initiative (OSI) - 
+	 * http://www.opensource.org/licenses/mit-license.php
+	 * You may use EITHER of these licenses to work with / distribute this source code.
```

Comment:
It sounds like Joshua Gentry is releasing his overall code as AGPL but carves out and preserves the LGPL/MIT license that Arah J. Leonard assigned to his functions.  Since I only copied Arah's functions then it should be ok?